src/Storages/Kafka/StorageKafka.cpp

```diff
@@ -279,6 +301,9 @@ Pipe StorageKafka::read(
     if (mv_attached)
         throw Exception(ErrorCodes::QUERY_NOT_ALLOWED, "Cannot read from StorageKafka with attached materialized views");
 
+    CurrentMetrics::Increment metric_increment{CurrentMetrics::KafkaDirectReads};
```

Comment:
"Number of direct selects from Kafka currently executing" -- but lifetime of this metric_increment object can be much less than the execution time of the select, is it ok?

Reply:
Also thought about that. Maybe profile event counter would bf e enough here.

From the other hand - if it hang / or slow it will be visible, and will explain why background read (MV) can't start.

Reply:
Ah, misunderstand, you. Read here actually don't read but returns sources which will be read later.
Indeed, you're right. Will just remove it (passing it properly seems too much effort, and is not clear if needed), maybe i will add something like "nr of busy customers" metric

Reply:
fixed